wallet-cli/src/command/pretty.rs

```diff
@@ -3,32 +3,45 @@ use std::fmt;
 use std::string::String;
 
 use blockchain::{genesis, normal, types, Block, SscProof};
-use wallet_crypto::{address, hash, tx, util::hex};
+use wallet_crypto::{address, cbor, config, hash, hdwallet, redeem, tx, util::hex};
 
 use ansi_term::Colour;
 
 // Constants for the fmt::Display instance
 static DISPLAY_INDENT_SIZE: usize = 4; // spaces
 static DISPLAY_INDENT_LEVEL: usize = 0; // beginning starts at zero
 
-type AST = Vec<(Key, Val)>;
+type AST<'a> = Vec<(Key<'a>, Val<'a>)>;
```

Comment:
While it is nice to optimise things by removing non necessary dynamic allocation, this is for pretty printing, it is okay if it takes a micro second for allocation. Keep this change, but don't pursue further, this is not the scope of this PR : we need block parsing/understanding for last week.